### ISSUE TITLE
KnownColors TypeInitializationException on Ubuntu

### ISSUE THREAD

**Issue by ByronAP** [zkweb-framework/ZKWeb.System.Drawing#7]
Unhandled Exception: System.TypeInitializationException: The type initializer for 'System.Drawing.KnownColors' threw an exception. ---> System.TypeInitializationException: The type initializer for 'System.Drawing.GDIPlus' threw an exception. ---> System.DllNotFoundException: Unable to load DLL 'gdiplus.dll': The specified module could not be found.
 (Exception from HRESULT: 0x8007007E)
   at System.Drawing.GDIPlus.GdiplusStartup(UInt64& token, GdiplusStartupInput& input, GdiplusStartupOutput& output)
   at System.Drawing.GDIPlus..cctor()
   --- End of inner exception stack trace ---
   at System.Drawing.GDIPlus.RunningOnWindows()
   at System.Drawing.KnownColors..cctor()
   --- End of inner exception stack trace ---
   at System.Drawing.KnownColors.FromKnownColor(KnownColor kc)
   at QRCoder.QRCode.GetGraphic(Int32 pixelsPerModule)
   at QrNetCore.Program.WriteQRImage(String data, Int32 ppm) in /home/raffael/MonoProjects/QrCoderNetcore/QrNetCore/Program.cs:line 23
   at QrNetCore.Program.Main(String[] args) in /home/raffael/MonoProjects/QrCoderNetcore/QrNetCore/Program.cs:line 10

**Comment by 303248153** [comment]
Please install libgdiplus first:
```
apt-get install libgdiplus
cd /usr/lib
ln -s libgdiplus.so gdiplus.dll
```

**Comment by codebude** [comment]
Hi @303248153 ,

I had the same problem and for me your fix worked very well. Thanks!

**Comment by johnknoop** [comment]
This is really unfortunate. I'm also getting this when creating a simple QR code. The proposed solution isn't so easy for me, because I dont have one single linux server where my application runs. I could make sure libgdiplus is pre-installed by some fiddling with my deploy script, but it doesn't feel like the cleanest solution.

According to [this](https://github.com/dotnet/corefx/issues/20325) page, they will add System.Drawing to .net Core. Will that remove the need for libgdiplus?

**Comment by codebude** [comment]
Hi @johnknoop ,

if they add System.Drawing to .NET Core I think the complete package ZKWeb.System.Drawing would become obsolete. But they have talked about adding System.Drawing for a long time. So I don't expect to see it (in its entirity) in the near future...

You say you have problems generating a Simple QR Code. Which library do you use? The original (codebude/QRCoder --> https://www.nuget.org/packages/QRCoder/ ) or the ZKWeb fork (https://github.com/zkweb-framework/QRCoder).

If you used ZKWeb version, you could try to use the original version. I added .NET Core 2.0 support some days ago. If you already tried this, you have one more option.

You could use "BitmapByteQRCode" or "PngByteQRCode" instead of "QRCode" class as renderer. (More information in the wiki --> https://github.com/codebude/QRCoder/wiki/Advanced-usage---QR-Code-renderers ) Those too classes don't use System.Drawing at all.

**Comment by johnknoop** [comment]
Hi

I'm using the original library, but I don't think I've updated it within the last few days. I'll try that!

Thanks

**Comment by johnknoop** [comment]
Btw, `PngByteQRCode` doesn't seem to be available in version 1.3.1.

**Comment by codebude** [comment]
Hi @johnknoop ,

sorry that's my fault. BytePngQRCode will come in 1.3.2. It's available, if you check out the current source from Github and compile it yourself. (It's already there, but it isn't pushed to NuGet by now.)
